controllers/workflowdef/json.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2023 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workflowdef
+
+import (
+	"context"
+	"encoding/json"
+	"path"
+	"strings"
+
+	"github.com/serverlessworkflow/sdk-go/v2/model"
+
+	"github.com/kiegroup/kogito-serverless-operator/api/metadata"
+	"github.com/kiegroup/kogito-serverless-operator/utils"
+
+	"errors"
+
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+
+	operatorapi "github.com/kiegroup/kogito-serverless-operator/api/v1alpha08"
+)
+
+// GetJSONWorkflow return a Kogito compliant JSON format workflow as bytearray give a specific workflow CR
+func GetJSONWorkflow(workflowCR *operatorapi.KogitoServerlessWorkflow, ctx context.Context) ([]byte, error) {
```

Comment:
just thinking out of loud here, instead using json here, wouldn't be better to use yaml?
When adding it to other places like the dockerfile and the applied label, it will eliminate the need of escaping the double quotes and can improve the readability.

wdyt?

@davidesalerno @ricardozanini @mahima1997 @radtriste

Reply:
Yes we can try, can you open a JIRA?  link it to this one: https://issues.redhat.com/browse/KOGITO-9177

Reply:
sure.